### PR TITLE
Run functional tests on CI

### PR DESCRIPTION
**Pull request by malukenho**

### CODE REVIEW

**Comment by szepeviktor** [comment]
...you mean unit tests

**Comment by szepeviktor** [comment]
Add this as the very first step to resolve dependency problems.

```yaml
      - name: "Set default PHP version"
        run: "sudo update-alternatives --set php /usr/bin/php7.4"

```

**Comment by malukenho** [comment]
We don't have unit tests for this specific project 😅

**Comment by szepeviktor** [comment]
You're pushing so hard GitHub gets red hot!
May I help?

File: .github/workflows/php.yml [hunk 1]
```diff
@@ -14,11 +14,19 @@ jobs:
...
+      uses: shivammathur/setup-php@v2
+      with:
+        php-version: '7.4'
+
     - name: Validate composer.json and composer.lock
       run: composer validate --strict
 
     - name: Install dependencies
-      run: composer update --prefer-dist --no-progress
+      run: composer install --prefer-dist --no-progress
```
Comment (szepeviktor) [inline .github/workflows/php.yml:26]:
```suggestion
      run: composer update --prefer-dist --no-progress
```

We do not have the lock file in the repo: https://github.com/codelicia/xulieta-json

**Comment by szepeviktor** [comment]
kép

Very bad news 😿 Upstream error?

File: .github/workflows/php.yml (hunk 1 above)
Comment (malukenho) [reply .github/workflows/php.yml:26]: The fact that we do not have the lock file on the repo means that we don't need to use `update` right?

**Comment by malukenho** [comment]
@szepeviktor yes. May you try with this fix https://github.com/codelicia/xulieta/pull/70?

File: .github/workflows/php.yml (hunk 1 above)
Comment (szepeviktor) [reply .github/workflows/php.yml:26]: Let Composer answer: https://getcomposer.org/doc/01-basic-usage.md#updating-dependencies-to-their-latest-versions

Comment (malukenho) [reply .github/workflows/php.yml:26]: Awesome @szepeviktor!

**Pull request merged**